react/modules/subscriptions.ts

```diff
@@ -0,0 +1,144 @@
+import { defineMessages, IntlShape } from 'react-intl'
+
+type SubscriptionFrequencyType = 'year' | 'month' | 'week' | 'day'
+type SubscriptionFrequency = {
+  interval: number
+  type: SubscriptionFrequencyType
+}
+
+const FREQUENCY_PATTERN = /(\d{1,3})\s*(\w*?)(?:$|s|ly)/i
+
+const SUBSCRIPTION_PREFIX = `vtex.subscription`
+const SUBSCRIPTION_KEY_PREFIX = `${SUBSCRIPTION_PREFIX}.key`
+
+export const SUBSCRIPTION_KEY_FREQUENCY = `${SUBSCRIPTION_KEY_PREFIX}.frequency`
+export const SUBSCRIPTION_KEY_PURCHASEDAY = `${SUBSCRIPTION_KEY_PREFIX}.purchaseday`
+
+const messages = defineMessages({
+  subscription: { id: 'store/product-customizer.subscription' },
+  frequency: { id: 'store/product-customizer.subscription.frequency' },
+  purchaseday: { id: 'store/product-customizer.subscription.purchaseday' },
+  purchaseDayLabel: {
+    id: 'store/product-customizer.subscription.purchaseday.day',
+  },
+})
+
+const frequencyMessages = defineMessages({
+  day: { id: 'store/product-customizer.subscription.frequency.day' },
+  month: { id: 'store/product-customizer.subscription.frequency.month' },
+  week: { id: 'store/product-customizer.subscription.frequency.week' },
+  year: { id: 'store/product-customizer.subscription.frequency.year' },
+})
+
+export function isSubscription(id: string) {
+  return Boolean(id.indexOf(SUBSCRIPTION_PREFIX) === 0)
+}
+
+function getSubscriptionMessageId(id: string) {
+  // uses the 'keyName' of vtex.subscription.key.{keyName}
+  if (id.indexOf(SUBSCRIPTION_KEY_PREFIX) === 0) {
+    return messages[
+      id.slice(SUBSCRIPTION_KEY_PREFIX.length + 1) as keyof typeof messages
+    ]
+  }
+
+  // uses 'subscriptions' for vtex.subscription.*
+  if (id.indexOf(SUBSCRIPTION_PREFIX) === 0) {
+    return messages.subscription
+  }
+
+  return { id }
+}
+
+export function formatSubscriptionLabel(id: string, intl: IntlShape) {
+  const messageId = getSubscriptionMessageId(id)
+
+  if (!messageId) return id
```

Comment:
Maybe use `formatIOMessage` in this case? I don't know how the user would save a different translation though 🤔 
Not sure if we should bother since it's not possible to customize this in teh Checkout as well.

Reply:
Would we gain something from switching to it?